pages/transactions.js

```diff
@@ -158,6 +173,30 @@ class TransactionsPage extends React.Component {
       this.setState({ Collective: currentCollective });
     }
 
+    const hasChildren =
+      (this.props.data?.transactions?.nodes || []).some(
+        el =>
+          el.fromAccount?.parent?.id === currentCollective.id ||
+          el.toAccount?.parent?.id === this.props.data?.account?.id,
+      ) || this.props.query.ignoreChildrenTransactions;
+    if (isNil(this.state.hasChildren) && hasChildren) {
+      this.setState({ hasChildren });
+    }
+
+    const hasGiftCards =
+      (this.props.data?.transactions?.nodes || []).some(el => el.giftCardEmitterAccount?.id) ||
+      this.props.query.ignoreGiftCardsTransactions;
+    if (isNil(this.state.hasGiftCards) && hasGiftCards) {
+      this.setState({ hasGiftCards });
+    }
+
+    const hasIncognito =
+      (this.props.data?.transactions?.nodes || []).some(el => el.account?.isIncognito) ||
+      this.props.query.ignoreIncognitoTransactions;
+    if (isNil(this.state.hasIncognito) && hasIncognito) {
+      this.setState({ hasIncognito });
+    }
```

Comment:
Ideally these properties would come from the backend, so we don't end up adding the checkbox as we paginate through transactions.

Reply:
True but good enough for now as backend cost/complexity would be much bigger.

Reply:
I'm okay having the checkbox appear while paginating.